scripts/interactive_tests/interactive_test_util.js

```diff
@@ -47,11 +52,16 @@ var Client = function(port) {
 
   // Connect to the server. 
   this.connect = function() {
-    var deferred = q.defer();
+    let deferredResolve;
```

Comment:
I think this could be simplified to

```
return new Promise((resolve, reject) => {
  socket = net.connect({port: port}, function() {
    resolve();
  })
})
```